learning_resources/etl/loaders.py

```diff
@@ -123,8 +123,10 @@ def load_instructors(
 def load_image(resource: LearningResource, image_data: dict) -> LearningResourceImage:
     """Load the image for a resource into the database"""
     if image_data:
-        image, _ = LearningResourceImage.objects.get_or_create(**image_data)
-
+        if LearningResourceImage.objects.filter(**image_data).exists():
+            image = LearningResourceImage.objects.filter(**image_data).first()
+        else:
+            image = LearningResourceImage.objects.create(**image_data)
```

Comment:
A potential drawback of this is that `LearningResourceImage.objects.filter(**image_data).first()` might return the wrong match.

For example an OCW website or different websites may have an `<img>` element in multiple places with the same url but different descriptions and/or alt text (and sometimes either of those might conceivably be null).

So if these have previously been created in the database:

```json
{'url':'http://baker.com', "description": "d1", "alt": "a1"}
{'url':'http://baker.com', "description": None, "alt": "a1"}
{'url':'http://baker.com', "description": "d1", "alt": None}
```

Then what `load_image(lr, image_data={'url':'http://baker.com'})` should do is create a new `LearningResourceImage` with  the same url and `description=None` and `alt=None`.  The PR code will return one of the other images instead.

So explicitly specifying all the possible image fields in the `get_or_create` call might be the way to go:

```python
image, _ = LearningResourceImage.objects.get_or_create(
    url=image_data.get("url"),
    description=image_data.get("description"),
    alt=image_data.get("alt"),
)
```

We could also ensure that all the ETL transform functions for the different sources populate every field for the image data even if it will always be null.  For example, in `learning_resources.etl.openedx._transform_image`:

```python
def _transform_image(image_data: dict) -> dict:
    """Return the transformed image data if a url is provided"""
    if image_data and image_data.get("src"):
        return {
            "url": image_data.get("src"),
            "description": image_data.get("description"),
            "alt": None # no alt value is ever included in the openedx api response
        }
    return None
```    

But this _might_ be overkill.

Reply:
Thats a good point. just pushed the change for using get_or_create with explicit fields